src/connectors/rating-menu/connectRatingMenu.js

```diff
@@ -129,39 +132,92 @@ export default function connectRatingMenu(renderFn, unmountFn = noop) {
   checkRendering(renderFn, withUsage());
 
   return (widgetParams = {}) => {
-    const { attribute, max = 5 } = widgetParams;
+    const { attribute, max = 5, step = 1 } = widgetParams;
     let sendEvent;
 
     if (!attribute) {
       throw new Error(withUsage('The `attribute` option is required.'));
     }
 
     const getRefinedStar = state => {
-      const refinements = state.getDisjunctiveRefinements(attribute);
+      const values = state.getNumericRefinements(attribute);
 
-      if (!refinements.length) {
+      if (!values['>=']?.length) {
         return undefined;
       }
 
-      return Math.min(...refinements.map(Number));
+      return values['>='][0];
+    };
+
+    const getFacetsMaxDecimalPlaces = facetResults => {
+      let maxDecimalPlaces = 0;
+      facetResults.forEach(facetResult => {
+        const [, decimal = ''] = facetResult.name.split('.');
+        maxDecimalPlaces = Math.max(maxDecimalPlaces, decimal.length);
+      });
+      return maxDecimalPlaces;
+    };
+
+    const getFacetValuesWarningMessage = ({
+      maxDecimalPlaces,
+      maxFacets,
+      maxValuesPerFacet,
+    }) => {
+      const maxDecimalPlacesInRange = Math.max(
+        0,
+        Math.floor(Math.log10(MAX_VALUES_PER_FACET_API_LIMIT / max))
+      );
+      const maxFacetsInRange = Math.min(
+        MAX_VALUES_PER_FACET_API_LIMIT,
+        Math.pow(10, maxDecimalPlacesInRange) * max
```

Comment:
the minimum could be different to 0, right?

Reply:
In theory yes, but the initial implementation was not handing this case. It is not supported in this implementation either.